src/include/sof/math/exp_fcn_hifi.h

```diff
@@ -0,0 +1,45 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ *
+ * Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+ *
+ */
+#ifndef __SOFM_EXP_HIFI_H__
+#define __SOFM_EXP_HIFI_H__
+
+#include <sof/platform.h>
+#include <stdint.h>
+#include <stddef.h>
+
+#undef EXPONENTIAL_GENERIC
+#undef EXPONENTIAL_HIFI3
+#undef EXPONENTIAL_HIFI4
+#undef EXPONENTIAL_HIFI5
+#if defined(__XCC__)
+
+#include <xtensa/config/core-isa.h>
+#if XCHAL_HAVE_HIFI5 == 1
+#define EXPONENTIAL_HIFI5 1
+
+#elif XCHAL_HAVE_HIFI4 == 1
+#define EXPONENTIAL_HIFI4 1
+
+#elif XCHAL_HAVE_HIFI3 == 1
+#define EXPONENTIAL_HIFI3 1
+#else
+#error "No HIFI3 found. Cannot build XCC EXP - HiFi module."
+#endif
+#else
+/* GCC */
+#if !defined(EXPONENTIAL_HIFI3) || !defined(EXPONENTIAL_HIFI4) || !defined(EXPONENTIAL_HIFI5)
```

Comment:
How could `EXPONENTIAL_HIFIx` be defined when `__XCC__` is not?

Reply:
So why are testing it if you know it's not defined?

Reply:
HiFi2 does not matter. Right now you have this code:

```C
#if defined(__XCC__)
...
#  define EXPONENTIAL_HIFI4
...
#else
...
#  if !defined(EXPONENTIAL_HIFI4) ...
...
#endif
```

EXPONENTIAL_HIFI4 is defined only in the top-level `#if` (as it should).  So of course it's not going to be defined in the top-level `#else`, testing it there makes no sense and is just super-confusing for the reader.